packages/blocks/google-cloud/src/lib/common-properties.ts

```diff
@@ -0,0 +1,55 @@
+import { Property } from '@openops/blocks-framework';
+import { runCommand } from './google-cloud-cli';
+
+type Project = {
+  name: string;
+  projectId: string;
+};
+
+export const projectCliDropdown = Property.Dropdown<string>({
+  displayName: 'Project',
+  description: 'Select a project to run the command in',
+  refreshers: [
+    'auth',
+    'useHostSession',
+    'useHostSession.useHostSessionCheckbox',
+  ],
+  required: true,
+  options: async ({ auth, useHostSession }) => {
+    if (!auth) {
+      return {
+        disabled: true,
+        options: [],
+        placeholder: 'Please authenticate to see projects.',
+      };
+    }
+
+    try {
+      const shouldUseHostCredentials =
+        (useHostSession as { useHostSessionCheckbox?: boolean })
+          ?.useHostSessionCheckbox === true;
+
+      const rawProjects = await runCommand(
+        'gcloud projects list --format=json',
+        auth,
+        shouldUseHostCredentials,
+      );
+
+      const projects: Project[] = JSON.parse(rawProjects) ?? [];
+
+      return {
+        disabled: false,
+        options: projects.map(({ name, projectId }) => ({
+          label: name,
+          value: projectId,
+        })),
+      };
+    } catch (error) {
+      return {
+        disabled: true,
+        options: [],
+        placeholder: `Error fetching projects: ${error}`,
```

Comment:
Alex added the "error" property, instead of using it in the placeholder 
https://github.com/openops-cloud/openops/blob/main/packages/blocks/azure/src/lib/common-properties.ts#L46